packages/backend/app/Http/Controllers/UsersController.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Contracts\Routing\ResponseFactory;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Validator;
+use Ramsey\Uuid\Uuid;
+
+class UsersController extends Controller
+{
+    public function signUp(Request $request): Response|JsonResponse|Application|ResponseFactory
+    {
+        $validator = Validator::make($request->all(), [
+            'name' => ['required', 'string', 'max:64'],
+            'email' => ['required', 'email', 'unique:users'],
+            'password' => ['required', 'string', 'min:8', 'max:32'],
+        ]);
+
+        if ($validator->fails()) {
+            return response(null, 400);
+        }
+
+        $uuid = Uuid::uuid7();
+
+        $user = new User();
+        $user->id = $uuid->toString();
+        $user->name = $validator->getData()['name'];
+        $user->email = $validator->getData()['email'];
+        $user->password = Hash::make($validator->getData()['password']);
+        $user->save();
+
+        $request->session()->regenerate();
+        Auth::login($user);
+
+        return response([
+            'id' => $user->id,
+            'name' => $user->name,
+            'email' => $user->email,
+        ], 201);
+    }
+
+    public function whoAmI(Request $request): Response|JsonResponse|Application|ResponseFactory
+    {
+        if ($request->user() === null) {
```

Comment:
[IMO]
組み込み関数を用いた方が比較対象が明示的になるため可読性が上がります。
```suggestion
        if (is_null($request->user())) {
```